File: backend/app/schemas/assumptions.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel

AssumptionSource = Literal["configured", "defaulted", "calculated"]


class AssumptionItemResponse(BaseModel):
    key: str
    label: str
    value: str
    source: AssumptionSource
    note: str | None = None
    warning: bool = False


class AssumptionSectionResponse(BaseModel):
    key: str
    title: str
    items: list[AssumptionItemResponse]


class ProjectAssumptionsResponse(BaseModel):
    project_id: UUID
    calculation_run_id: UUID | None
    scenario_name: str | None
    engine_version: str
    generated_at: datetime | None
    warnings: list[str]
    sections: list[AssumptionSectionResponse]

File: frontend/types/assumptions.ts
export type AssumptionSource = "configured" | "defaulted" | "calculated";

export type AssumptionItem = {
  key: string;
  label: string;
  value: string;
  source: AssumptionSource;
  note: string | null;
  warning: boolean;
};

export type AssumptionSection = {
  key: string;
  title: string;
  items: AssumptionItem[];
};

export type ProjectAssumptions = {
  project_id: string;
  calculation_run_id: string | null;
  scenario_name: string | null;
  engine_version: string;
  generated_at: string | null;
  warnings: string[];
  sections: AssumptionSection[];
};
